docs/design/proposals/destructive-operations/destructive-operations.md

```diff
@@ -0,0 +1,202 @@
+# Semantics of destructive operations
+
+## Summary
+
+This design document stems from the initial discussion on the [original Github issue #82](https://github.com/aws-controllers-k8s/community/issues/82).
+
+ACK currently treats all resources with the same amount of protection for deletion. That is, if you delete an ACK custom resource, the underlying AWS resource is deleted as well. While this might be the expected behaviour for the vast majority of resources, it also increases the chance that an AWS resource is accidentally deleted - through prematurely or accidentally deleting the ACK CR. Some sort of deletion protection should be introduced to limit the odds of this happening, requiring the user to unset a property manually before the controller will continue with deletion.
```

Comment:
> Some sort of deletion protection should be introduced to limit the odds of this happening, requiring the user to unset a property manually before the controller will continue with deletion.

Honestly, I really don't care about this "use case". It's so very imperative and GUI/console experience-centric. There is a reason we override all the `--dry-run` and `--skip-final-snapshot` attributes in our ACK controllers, and that is because we cater to an automated, declarative configuration management strategy. I don't really want to replicate the giant mess of "type `delete` here" (and myriad variations of this) AWS console dialog boxes in the ACK user experience.

On the other hand, I am 100% supportive of the opposite direction of thought: facilitating an **easy** and **consistent** way of deleting *child/dependent resources* when a parent resource has been deleted -- in other words, supporting a consistent cascading deletion experience for all AWS API resources.

Note that I do *not* consider a Kubernetes Namespace to be the parent of any ACK resource. In other words, deleting a Kubernetes Namespace should not delete all ACK CRs that happen to be in that Namespace. The Namespace should not be able to be deleted until all CRs or other Kubernetes-native resources in that Namespace are deleted themselves.